Sources/Subscription/Services/SubscriptionService.swift

```diff
@@ -55,7 +56,9 @@ public final class SubscriptionService: APIService {
 
         switch result {
         case .success(let subscriptionResponse):
-            subscriptionCache.set(subscriptionResponse)
+            let defaultExpiryDate = Date().addingTimeInterval(subscriptionCache.settings.defaultExpirationInterval)
```

Comment:
This relies on system time which can be manipulated. If device time is set to a distance future, any entitlement check that depends on the cache would just succeed. While we do use `reloadIgnoringLocalCacheData` when refreshing periodically, we don't currently handle the failure.